File: c/practice/binary-search.c
#include<stdio.h>
int main(){
    int arr[5],low=0,mid,high=4,key,found=0;
    printf("enter the elements:\n");
    for(int i=0;i<5;i++){
        scanf("%d",&arr[i]);
    }
    printf("enter the element to be searched:\n");
    scanf("%d",&key);
    while(low<=high){
        mid=(low+high)/2;
        if(arr[mid]==key){
            printf("key element %d is found at index %d\n",key,mid);
            found=1;
            break;
        }
        else if(arr[mid]<key){
            low=mid+1;
        }
        else{
            high=mid-1;
        }
    

    }
    if(!found){
        printf("key element not found in the array\n");

    }
    return 0;
}

File: Eclipse workspace.java/practicecode/src/arrays/BinarySearch.java
package arrays;

import java.util.Scanner;

public class BinarySearch {
	public static int BinarySearch(int[] arr, int target) {
//		int low = 0, high = arr.length - 1;
//		while (low <= high) {
//
//			int mid = low + (high - low) / 2;
//			if (arr[mid] == target) {
//
//				return mid;
//			} else if (arr[mid] < target) {
//				low = mid + 1;
//
//			} else {
//				high = mid - 1;
//			}
//		}
//
//		return low;
		int low = 0, high = arr.length - 1;
		while (low <= high) {
			int mid = low + (high - low) / 2;
			if (arr[mid] == target) {
				return mid;
			} else if (arr[mid] < target) {
				low = mid + 1;
			} else {
				high = mid - 1;
			}
		}
		return low;
	}

	public static void main(String[] args) {
//		Scanner sc = new Scanner(System.in);
//		System.out.println("enter the size of an array:");
//		int size = sc.nextInt();
//		System.out.println("enter the elements:");
//		int[] arr = new int[size];
//		for (int i = 0; i < size; i++) {
//			arr[i] = sc.nextInt();
//		}
//		System.out.println("enter target:");
//		int target = sc.nextInt();
//		int result = BinarySearch(arr, target);
//		System.out.println(result);
		Scanner sc = new Scanner(System.in);
		System.out.println("enter the size of an array:");
		int size = sc.nextInt();
		int[] arr = new int[size];
		System.out.println("enter the elements:");
		for (int i = 0; i < size; i++) {
			arr[i] = sc.nextInt();
		}
		System.out.println("enter the target");
		int target = sc.nextInt();
		int result = BinarySearch(arr, target);
		System.out.println(result);
	}

}
